test/test_sequence.py

```diff
@@ -0,0 +1,119 @@
+from test.testbase import DbIsolatedTest, run_test_as_sync
+
+import random
+from sample_metadata.models import (
+    SequenceType,
+    SequenceStatus,
+    SampleType,
+)
+from db.python.layers.sample import SampleLayer
+from db.python.layers.sequence import SampleSequenceLayer
+
+
+class TestSequence(DbIsolatedTest):
+    """Test sequence class"""
+
+    @run_test_as_sync
+    async def setUp(self) -> None:
+        super().setUp()
+
+        self.sl = SampleLayer(self.connection)
+        self.seql = SampleSequenceLayer(self.connection)
+        self.external_sample_id = 'TESTING001'
+        self.sequence_type = 'genome'
+        self.sequence_status = 'unknown'
+
+        # Create new sample
+        await self.sl.insert_sample(
+            self.external_sample_id,
+            SampleType('blood'),
+            active=True,
+            meta={'Testing': 'test_sequence.py'},
+        )
+
+        sample_ids = await self.sl.get_sample_id_map_by_external_ids(
+            [self.external_sample_id], project=None
+        )
+        self.sample_id = sample_ids[self.external_sample_id]
+
+        # Create new sequence
+        _ = await self.seql.insert_sequencing(
+            self.sample_id,
+            SequenceType(self.sequence_type),
+            SequenceStatus(self.sequence_status),
+            {},
+        )
+
+    @run_test_as_sync
+    async def test_update_sequence_from_sample_and_type(self):
```

Comment:
Could you add a test for no sample present, and sample present but no sequence available?

Reply:
Done!